src/megaclient.cpp

```diff
@@ -12983,8 +12983,8 @@ error MegaClient::isLocalPathSyncable(const LocalPath& newPath, handle excludeBa
 
     error e = API_OK;
     syncs.forEachSyncConfig([&](const SyncConfig& config){
-
-        if (config.getBackupId() != excludeBackupId && excludeBackupId != UNDEF) // for the check inside addsync() when it's already present
+        // For the check inside addsync(...) when the config's already present.
+        if (excludeBackupId == UNDEF || config.getBackupId() != excludeBackupId)
```

Comment:
Good catch, David. This is another bug introduced by the revamping of the sync-management & sync-heartbeats to pre-register syncs in the API before they are effectively added to the set.

Given that no sync-config can be added without a backupId, I'd say that we can simplify the check. I'd also make a more detailed comment:
```suggestion cpp
        // skip the check if the sync has not been registered yet (check from addsync() -> excludeBackupId = UNDEF)...
        // or if the sync has been registered already (check from eanblesync() -> excludeBackupId is this config's id)
        if (config.getBackupId() != excludeBackupId)
```